### PR TITLE
removed scheduler and use to mongo db to save the notify the scheduler

### PR DESCRIPTION
**Pull request by sfahad1414**
## Summary by CodeRabbit

- **New Features**
  - Enhanced scheduling capabilities with a new asynchronous method for adding scheduled jobs.
  - Improved handling of PDF documents using an updated library for better performance.
  - Streamlined introduction of Kairon in the README, focusing on its functionalities.
  - Updated WhatsApp API integration to the latest version, ensuring compatibility with new features.

- **Bug Fixes**
  - Updated dependencies to the latest versions, potentially fixing bugs and improving performance.

- **Tests**
  - Restructured test cases for action scheduling, introducing new tests for improved coverage and validation.
  - Updated test cases to align with the new WhatsApp API version.

- **Chores**
  - Updated library dependencies to ensure the application utilizes the latest features and fixes.

### CODE REVIEW

**Review by hiteshghuge — DISMISSED** [review]
reviewed

File: tests/unit_test/callback_test.py [hunk 1]
```diff
@@ -272,14 +272,14 @@
...
-    config = {'access_token': 'dummy_token'}
+    config = {'access_token': '270485614:AAHfiqksKZ8WmR2zSjiQ7_v4TMAKdiHm9T0'}
     with patch('kairon.chat.handlers.channels.telegram.TelegramOutput.send_text_message', new_callable=AsyncMock) as mock_send_message:
         await ChannelMessageDispatcher.handle_telegram('bot', config, 'sender', 'text message')
         mock_send_message.assert_called_once_with('sender', 'text message')
 
 @pytest.mark.asyncio
 async def test_handle_telegram_custom_json():
-    config = {'access_token': 'dummy_token'}
+    config = {'access_token': '270485614:AAHfiqksKZ8WmR2zSjiQ7_v4TMAKdiHm9T0'}
```
[the remark this answers is not in the document]
Comment (sfahad1414) [reply tests/unit_test/callback_test.py:282]: this is the dummy token

**Review by hiteshghuge — DISMISSED** [review]
approved

**Review by hiteshghuge — APPROVED** [review]
reviewed

**Pull request closed without merging**